File: software/lamp-ui/src/utils/stringCase.test.ts
import { describe, expect, it } from 'vitest'
import { toTitleCase } from './stringCase'

describe('toTitleCase', () => {
  it('single lowercase word', () => expect(toTitleCase('jacko')).toBe('Jacko'))
  it('multi word', () => expect(toTitleCase('living room')).toBe('Living Room'))
  it('already capitalized normalizes', () => expect(toTitleCase('LIVING ROOM')).toBe('Living Room'))
  it('empty stays empty', () => expect(toTitleCase('')).toBe(''))
  it('blank whitespace preserved', () => expect(toTitleCase('   ')).toBe('   '))
})

File: software/lamp-app-flutter/test/core/utils/string_case_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:lamp_app/core/utils/string_case.dart';

void main() {
  group('toTitleCase', () {
    test('single lowercase word', () {
      expect('jacko'.toTitleCase(), 'Jacko');
    });

    test('multi word', () {
      expect('living room'.toTitleCase(), 'Living Room');
    });

    test('already capitalized normalizes', () {
      expect('LIVING ROOM'.toTitleCase(), 'Living Room');
    });

    test('empty stays empty', () {
      expect(''.toTitleCase(), '');
    });

    test('blank whitespace preserved', () {
      expect('   '.toTitleCase(), '   ');
    });
  });
}
